src/EditorFeatures/CSharpTest2/Recommendations/StructKeywordRecommenderTests.cs

```diff
@@ -521,5 +521,168 @@ public async Task TestNotAfterRecord_SealedModifier()
         {
             await VerifyAbsenceAsync("sealed record $$");
         }
+
+        [Fact]
+        public async Task TestAfterAllowsRefInTypeParameterConstraint()
+        {
+            await VerifyKeywordAsync(
+@"class C<T> where T : allows ref $$");
+        }
+
+        [Fact]
+        public async Task TestAfterAllowsRefInTypeParameterConstraint2()
+        {
+            await VerifyKeywordAsync(
+                """
+                class C<T>
+                    where T : allows ref $$
+                    where U : U
+                """);
+        }
+
+        [Fact]
+        public async Task TestAfterAllowsRefInMethodTypeParameterConstraint()
+        {
+            await VerifyKeywordAsync(
+                """
+                class C {
+                    void Goo<T>()
+                      where T : allows ref $$
+                """);
+        }
+
+        [Fact]
+        public async Task TestAfterAllowsRefInMethodTypeParameterConstraint2()
+        {
+            await VerifyKeywordAsync(
+                """
+                class C {
+                    void Goo<T>()
+                      where T : allows ref $$
+                      where U : T
+                """);
+        }
+
+        [Fact]
+        public async Task TestAfterAllowsRefAfterClassTypeParameterConstraint()
+        {
+            await VerifyKeywordAsync(
+@"class C<T> where T : class, allows ref $$");
```

Comment:
could these all be raw strings plz.  or indent 4 from `await`.  we're trying to move away entirely from tests that are 0-column aligned.